pilot/pkg/xds/debug.go

```diff
@@ -293,6 +297,8 @@ func (s *DiscoveryServer) endpointz(w http.ResponseWriter, req *http.Request) {
 			all := s.Env.ServiceDiscovery.InstancesByPort(ss, p.Port, nil)
 			_, _ = fmt.Fprintf(w, "\n{\"svc\": \"%s:%s\", \"ep\": [\n", ss.Hostname, p.Name)
 			for _, svc := range all {
+				// prevent crash during serialization
+				svc = svc.DeepCopy()
```

Comment:
why not use the mutex like above?

Reply:
This `svc` is actually a `ServiceInstance`, which embeds a `Service` object, so we'd be doing something like `svc.Service.RLock()`, and because I don't know what other nested fields in `ServiceInstance` might cause problems, it seemed best to just copy it before serializing it.